From bb32dd981999b81d9796b90ce4461b654f0f122c Mon Sep 17 00:00:00 2001
From: HardevSinh Palaniya <hardevsinh.palaniya@siliconsignals.io>
Date: Fri, 28 Apr 2023 18:43:18 +0530
Subject: [PATCH] Add fuel gauge support

- Apply changes in dts
- modify i2c node

Signed-off-by: HardevSinh Palaniya <hardevsinh.palaniya@siliconsignals.io>

diff --git a/arch/arm64/boot/dts/freescale/imx8mm-mecha-comet-m-gen1.dts b/arch/arm64/boot/dts/freescale/imx8mm-mecha-comet-m-gen1.dts
index d827acdd9532..4324c0ec7ef7 100644
--- a/arch/arm64/boot/dts/freescale/imx8mm-mecha-comet-m-gen1.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mm-mecha-comet-m-gen1.dts
@@ -7,3 +7,79 @@
 
 #include "imx8mm-mecha-som-gen1.dts"
 
+&i2c2 {
+	clock-frequency = <400000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c2>;
+	status = "okay";
+};
+
+&i2c3 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3>;
+	status = "okay";
+
+	bq27440: fuel-gauge@55 {
+		compatible = "ti,bq27441";
+		reg = <0x55>;
+		monitored-battery = <&bat>;
+		bat: battery {
+			compatible = "simple-battery";
+			over-voltage-threshold-microvolt = <4500000>;
+			re-charge-voltage-microvolt = <250000>;
+			voltage-min-design-microvolt = <3200000>;
+			voltage-max-design-microvolt = <4200000>;
+			energy-full-design-microwatt-hours = <5290000>;
+			charge-full-design-microamp-hours = <1430000>;
+			precharge-current-microamp = <256000>;
+			precharge-upper-limit-microvolt = <2500000>;
+			charge-term-current-microamp = <128000>;
+			constant-charge-current-max-microamp = <900000>;
+			constant-charge-voltage-max-microvolt = <4200000>;
+			factory-internal-resistance-micro-ohms = <250000>;
+			ocv-capacity-celsius = <(-10) 0 10>;
+			/* table for -10 degree Celsius */
+			ocv-capacity-table-0 = <4185000 100>, <4113000 95>, <4066000 90>;
+			/* table for 0 degree Celsius */
+			ocv-capacity-table-1 = <4200000 100>, <4185000 95>, <4113000 90>;
+			/* table for 10 degree Celsius */
+			ocv-capacity-table-2 = <4250000 100>, <4200000 95>, <4185000 90>;
+			resistance-temp-table = <20 100>, <10 90>, <0 80>, <(-10) 60>;
+			operating-range-celsius = <(-30) 50>;
+			ambient-celsius = <(-5) 50>;
+			alert-celsius = <0 40>;
+		};
+	};
+
+};
+
+&i2c4 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c4>;
+	status = "okay";
+};
+
+&iomuxc {
+	pinctrl_i2c2: i2c2grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_I2C2_SCL_I2C2_SCL                          0x400001c3
+			MX8MM_IOMUXC_I2C2_SDA_I2C2_SDA                          0x400001c3
+		>;
+	};
+
+	pinctrl_i2c3: i2c3grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_I2C3_SCL_I2C3_SCL                          0x400001c3
+			MX8MM_IOMUXC_I2C3_SDA_I2C3_SDA                          0x400001c3
+		>;
+	};
+
+	pinctrl_i2c4: i2c4grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_I2C4_SCL_I2C4_SCL                          0x400001c3
+			MX8MM_IOMUXC_I2C4_SDA_I2C4_SDA                          0x400001c3
+		>;
+	};
+};
-- 
2.25.1

