From 4d6ded671e28fbeceefba2198b162d3983e3a18d Mon Sep 17 00:00:00 2001
From: HardevSinh Palaniya <hardevsinh.palaniya@siliconsignals.io>
Date: Fri, 5 May 2023 14:42:29 +0530
Subject: [PATCH] Added panel-sitronix-st7701 support


diff --git a/drivers/gpu/drm/panel/panel-sitronix-st7701.c b/drivers/gpu/drm/panel/panel-sitronix-st7701.c
index 320a2a8fd459..dd21170b83e6 100644
--- a/drivers/gpu/drm/panel/panel-sitronix-st7701.c
+++ b/drivers/gpu/drm/panel/panel-sitronix-st7701.c
@@ -1,418 +1,611 @@
-// SPDX-License-Identifier: GPL-2.0+
 /*
- * Copyright (C) 2019, Amarula Solutions.
- * Author: Jagan Teki <jagan@amarulasolutions.com>
+ * i.MX drm driver - Raydium MIPI-DSI panel driver
+ *
+ * Copyright (C) 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
  */
 
+
+#include <drm/drm_crtc.h>
 #include <drm/drm_mipi_dsi.h>
-#include <drm/drm_modes.h>
 #include <drm/drm_panel.h>
-
 #include <linux/gpio/consumer.h>
-#include <linux/delay.h>
 #include <linux/module.h>
-#include <linux/of_device.h>
+#include <linux/of.h>
+#include <linux/printk.h>
 #include <linux/regulator/consumer.h>
-
 #include <video/mipi_display.h>
+#include <video/of_videomode.h>
+#include <video/videomode.h>
+#include <linux/backlight.h>
+#include <drm/drm_print.h>
+#include <linux/delay.h>
 
-/* Command2 BKx selection command */
-#define DSI_CMD2BKX_SEL			0xFF
-
-/* Command2, BK0 commands */
-#define DSI_CMD2_BK0_PVGAMCTRL		0xB0 /* Positive Voltage Gamma Control */
-#define DSI_CMD2_BK0_NVGAMCTRL		0xB1 /* Negative Voltage Gamma Control */
-#define DSI_CMD2_BK0_LNESET		0xC0 /* Display Line setting */
-#define DSI_CMD2_BK0_PORCTRL		0xC1 /* Porch control */
-#define DSI_CMD2_BK0_INVSEL		0xC2 /* Inversion selection, Frame Rate Control */
-
-/* Command2, BK1 commands */
-#define DSI_CMD2_BK1_VRHS		0xB0 /* Vop amplitude setting */
-#define DSI_CMD2_BK1_VCOM		0xB1 /* VCOM amplitude setting */
-#define DSI_CMD2_BK1_VGHSS		0xB2 /* VGH Voltage setting */
-#define DSI_CMD2_BK1_TESTCMD		0xB3 /* TEST Command Setting */
-#define DSI_CMD2_BK1_VGLS		0xB5 /* VGL Voltage setting */
-#define DSI_CMD2_BK1_PWCTLR1		0xB7 /* Power Control 1 */
-#define DSI_CMD2_BK1_PWCTLR2		0xB8 /* Power Control 2 */
-#define DSI_CMD2_BK1_SPD1		0xC1 /* Source pre_drive timing set1 */
-#define DSI_CMD2_BK1_SPD2		0xC2 /* Source EQ2 Setting */
-#define DSI_CMD2_BK1_MIPISET1		0xD0 /* MIPI Setting 1 */
+u8 dcs0[] = {0xFF ,0x77 ,0x01 ,0x00 ,0x00 ,0x13};
+u8 dcs1[] = {0xEF, 0x08};
+u8 dcs2[] = {0xFF, 0x77, 0x01, 0x00, 0x00, 0x10};
+u8 dcs3[] = {0xC0, 0xE9, 0x03};
+u8 dcs4[] = {0xC1, 0x10, 0x0C};
+u8 dcs5[] = {0xC2, 0x01, 0x0A};
+u8 dcs6[] = {0xC3, 0x02};
+u8 dcs7[] = {0xCC, 0x10};
+u8 dcs8[] = {0xCD, 0x08};
+u8 dcs9[] =  {0xB0, 0x0D, 0x14, 0x9C, 0x0B, 0x10, 0x06, 0x08, 0x09, 0x08, 0x22, 0x02, 0x4F, 0x0E, 0x66, 0x2D, 0x1F };
+u8 dcs10[] = {0xB1, 0x00, 0x17, 0x9E, 0x0F, 0x11, 0x06, 0x0C, 0x08, 0x08, 0x26, 0x04, 0x51, 0x10, 0x6A, 0x33, 0x1B };
+u8 dcs11[] = {0xFF, 0x77, 0x01, 0x00, 0x00, 0x11 };
+u8 dcs12[] = {0xB0, 0x30 };
+u8 dcs13[] = {0xB1, 0x67};
+u8 dcs14[] = {0xB2, 0x84};
+u8 dcs15[] = {0xB3, 0x80 };
+u8 dcs16[] = {0xB5, 0x4E };
+u8 dcs17[] = {0xB7, 0x85 };
+u8 dcs18[] = {0xB8, 0x20 };
+u8 dcs19[] = {0xC1, 0x78 };
+u8 dcs20[] = {0xC2, 0x78 };
+u8 dcs21[] = {0xD0, 0x88 };
+u8 dcs22[] = {0xE0, 0x00, 0x00, 0x02 };
+u8 dcs23[] = {0xE1, 0x06, 0xA0, 0x08, 0xA0, 0x05, 0xA0, 0x07, 0xA0, 0x00, 0x44, 0x44};
+u8 dcs24[] = {0xE2, 0x30, 0x30, 0x44, 0x44, 0x6E, 0xA0, 0x00, 0x00, 0x6E, 0xA0, 0x00, 0x00 };
+u8 dcs25[] = {0xE3, 0x00, 0x00, 0x33, 0x33 };
+u8 dcs26[] = {0xE4, 0x44, 0x44 };
+u8 dcs27[] = {0xE5, 0x0D, 0x69, 0x0A, 0xA0, 0x0F, 0x6B, 0x0A, 0xA0, 0x09, 0x65, 0x0A, 0xA0, 0x0B, 0x67, 0x0A, 0xA0 };
+u8 dcs28[] = {0xE6, 0x00, 0x00, 0x33, 0x33 };
+u8 dcs29[] = {0xE7, 0x44, 0x44 };
+u8 dcs30[] = {0xE8, 0x0C, 0x68, 0x0A, 0xA0, 0x0E, 0x6A, 0x0A, 0xA0, 0x08, 0x64, 0x0A, 0xA0, 0x0A, 0x66, 0x0A, 0xA0 };
+u8 dcs31[] = {0xE9, 0x36 , 0x00 };
+u8 dcs32[] = {0xEB, 0x00, 0x01, 0xE4, 0xE4, 0x44, 0x88, 0x40 };
+u8 dcs33[] = {0xED, 0xFF, 0x45, 0x67, 0xFA, 0x01, 0x2B, 0xCF, 0xFF, 0xFF, 0xFC, 0xB2, 0x10, 0xAF, 0x76, 0x54, 0xFF };
+u8 dcs34[] = {0xEF, 0x10, 0x0D, 0x04, 0x08, 0x3F, 0x1F };
+u8 dcs35[] = {0x3A, 0x66 };
+u8 dcs36[] = {0x11, 0x00};
+u8 dcs37[] = {0x29, 0x00};
+u8 dcs38[] = {0x35, 0x00};
+
+unsigned char *ST7701S_CMD_Group[]={
+	dcs0,
+	dcs1,
+	dcs2,
+	dcs3,
+	dcs4,
+	dcs5,
+	dcs6,
+	dcs7,
+	dcs8,
+	dcs9,
+	dcs10,
+	dcs11,
+	dcs12,
+	dcs13,
+	dcs14,
+	dcs15,
+	dcs16,
+	dcs17,
+	dcs18,
+	dcs19,
+	dcs20,
+	dcs21,
+	dcs22,
+	dcs23,
+	dcs24,
+	dcs25,
+	dcs26,
+	dcs27,
+	dcs28,
+	dcs29,
+	dcs30,
+	dcs31,
+	dcs32,
+	dcs33,
+	dcs34,
+	dcs35,
+	dcs36,
+	dcs37,
+	dcs38,
+};
+unsigned long int ST7701S_CMD_Group_size[]={
+	sizeof(dcs0),
+	sizeof(dcs1),
+	sizeof(dcs2),
+	sizeof(dcs3),
+	sizeof(dcs4),
+	sizeof(dcs5),
+	sizeof(dcs6),
+	sizeof(dcs7),
+	sizeof(dcs8),
+	sizeof(dcs9),
+	sizeof(dcs10),
+	sizeof(dcs11),
+	sizeof(dcs12),
+	sizeof(dcs13),
+	sizeof(dcs14),
+	sizeof(dcs15),
+	sizeof(dcs16),
+	sizeof(dcs17),
+	sizeof(dcs18),
+	sizeof(dcs19),
+	sizeof(dcs20),
+	sizeof(dcs21),
+	sizeof(dcs22),
+	sizeof(dcs23),
+	sizeof(dcs24),
+	sizeof(dcs25),
+	sizeof(dcs26),
+	sizeof(dcs27),
+	sizeof(dcs28),
+	sizeof(dcs29),
+	sizeof(dcs30),
+	sizeof(dcs31),
+	sizeof(dcs32),
+	sizeof(dcs33),
+	sizeof(dcs34),
+	sizeof(dcs35),
+	sizeof(dcs36),
+	sizeof(dcs37),
+	sizeof(dcs38),
+};
 
-/*
- * Command2 with BK function selection.
- *
- * BIT[4, 0]: [CN2, BKXSEL]
- * 10 = CMD2BK0, Command2 BK0
- * 11 = CMD2BK1, Command2 BK1
- * 00 = Command2 disable
- */
-#define DSI_CMD2BK1_SEL			0x11
-#define DSI_CMD2BK0_SEL			0x10
-#define DSI_CMD2BKX_SEL_NONE		0x00
-
-/* Command2, BK0 bytes */
-#define DSI_LINESET_LINE		0x69
-#define DSI_LINESET_LDE_EN		BIT(7)
-#define DSI_LINESET_LINEDELTA		GENMASK(1, 0)
-#define DSI_CMD2_BK0_LNESET_B1		DSI_LINESET_LINEDELTA
-#define DSI_CMD2_BK0_LNESET_B0		(DSI_LINESET_LDE_EN | DSI_LINESET_LINE)
-#define DSI_INVSEL_DEFAULT		GENMASK(5, 4)
-#define DSI_INVSEL_NLINV		GENMASK(2, 0)
-#define DSI_INVSEL_RTNI			GENMASK(2, 1)
-#define DSI_CMD2_BK0_INVSEL_B1		DSI_INVSEL_RTNI
-#define DSI_CMD2_BK0_INVSEL_B0		(DSI_INVSEL_DEFAULT | DSI_INVSEL_NLINV)
-#define DSI_CMD2_BK0_PORCTRL_B0(m)	((m)->vtotal - (m)->vsync_end)
-#define DSI_CMD2_BK0_PORCTRL_B1(m)	((m)->vsync_start - (m)->vdisplay)
-
-/* Command2, BK1 bytes */
-#define DSI_CMD2_BK1_VRHA_SET		0x45
-#define DSI_CMD2_BK1_VCOM_SET		0x13
-#define DSI_CMD2_BK1_VGHSS_SET		GENMASK(2, 0)
-#define DSI_CMD2_BK1_TESTCMD_VAL	BIT(7)
-#define DSI_VGLS_DEFAULT		BIT(6)
-#define DSI_VGLS_SEL			GENMASK(2, 0)
-#define DSI_CMD2_BK1_VGLS_SET		(DSI_VGLS_DEFAULT | DSI_VGLS_SEL)
-#define DSI_PWCTLR1_AP			BIT(7) /* Gamma OP bias, max */
-#define DSI_PWCTLR1_APIS		BIT(2) /* Source OP input bias, min */
-#define DSI_PWCTLR1_APOS		BIT(0) /* Source OP output bias, min */
-#define DSI_CMD2_BK1_PWCTLR1_SET	(DSI_PWCTLR1_AP | DSI_PWCTLR1_APIS | \
-					DSI_PWCTLR1_APOS)
-#define DSI_PWCTLR2_AVDD		BIT(5) /* AVDD 6.6v */
-#define DSI_PWCTLR2_AVCL		0x0    /* AVCL -4.4v */
-#define DSI_CMD2_BK1_PWCTLR2_SET	(DSI_PWCTLR2_AVDD | DSI_PWCTLR2_AVCL)
-#define DSI_SPD1_T2D			BIT(3)
-#define DSI_CMD2_BK1_SPD1_SET		(GENMASK(6, 4) | DSI_SPD1_T2D)
-#define DSI_CMD2_BK1_SPD2_SET		DSI_CMD2_BK1_SPD1_SET
-#define DSI_MIPISET1_EOT_EN		BIT(3)
-#define DSI_CMD2_BK1_MIPISET1_SET	(BIT(7) | DSI_MIPISET1_EOT_EN)
-
-struct st7701_panel_desc {
-	const struct drm_display_mode *mode;
-	unsigned int lanes;
-	unsigned long flags;
-	enum mipi_dsi_pixel_format format;
-	const char *const *supply_names;
-	unsigned int num_supplies;
-	unsigned int panel_sleep_delay;
+static const u32 st7701s_bus_formats[] = {
+	MEDIA_BUS_FMT_RGB888_1X24,
+	MEDIA_BUS_FMT_RGB666_1X18,
+	MEDIA_BUS_FMT_RGB565_1X16,
 };
 
-struct st7701 {
-	struct drm_panel panel;
+ struct st7701s_panel {
+	struct drm_panel base;
 	struct mipi_dsi_device *dsi;
-	const struct st7701_panel_desc *desc;
 
-	struct regulator_bulk_data *supplies;
 	struct gpio_desc *reset;
-	unsigned int sleep_delay;
+	struct backlight_device *backlight;
+
+	bool prepared;
+	bool enabled;
+
+	struct videomode vm;
+	u32 width_mm;
+	u32 height_mm;
 };
 
-static inline struct st7701 *panel_to_st7701(struct drm_panel *panel)
+static inline struct st7701s_panel *to_st7701s_panel(struct drm_panel *panel)
 {
-	return container_of(panel, struct st7701, panel);
+	return container_of(panel, struct st7701s_panel, base);
 }
 
-static inline int st7701_dsi_write(struct st7701 *st7701, const void *seq,
-				   size_t len)
+static int st7701s_panel_push_cmd_list(struct mipi_dsi_device *dsi)
 {
-	return mipi_dsi_dcs_write_buffer(st7701->dsi, seq, len);
-}
-
-#define ST7701_DSI(st7701, seq...)				\
-	{							\
-		const u8 d[] = { seq };				\
-		st7701_dsi_write(st7701, d, ARRAY_SIZE(d));	\
+	size_t i;
+	int ret = 0;
+	struct device *dev = &dsi->dev;
+
+	dsi->mode_flags |= MIPI_DSI_MODE_LPM;
+
+	//send init cmds
+	for(i=0; i<39; i++)
+	{
+		ret = mipi_dsi_generic_write(dsi, ST7701S_CMD_Group[i],
+						ST7701S_CMD_Group_size[i]);
+		if (ret < 0) {
+			DRM_DEV_ERROR(dev, "Failed: CMD = 0x%x , length = %ld .\n",
+				ST7701S_CMD_Group[i][0], sizeof(ST7701S_CMD_Group[i]));
+			return ret;
+		}
 	}
 
-static void st7701_init_sequence(struct st7701 *st7701)
+	return ret;
+};
+
+static int st7701s_panel_prepare(struct drm_panel *panel)
 {
-	const struct drm_display_mode *mode = st7701->desc->mode;
-
-	ST7701_DSI(st7701, MIPI_DCS_SOFT_RESET, 0x00);
-
-	/* We need to wait 5ms before sending new commands */
-	msleep(5);
-
-	ST7701_DSI(st7701, MIPI_DCS_EXIT_SLEEP_MODE, 0x00);
-
-	msleep(st7701->sleep_delay);
-
-	/* Command2, BK0 */
-	ST7701_DSI(st7701, DSI_CMD2BKX_SEL,
-		   0x77, 0x01, 0x00, 0x00, DSI_CMD2BK0_SEL);
-	ST7701_DSI(st7701, DSI_CMD2_BK0_PVGAMCTRL, 0x00, 0x0E, 0x15, 0x0F,
-		   0x11, 0x08, 0x08, 0x08, 0x08, 0x23, 0x04, 0x13, 0x12,
-		   0x2B, 0x34, 0x1F);
-	ST7701_DSI(st7701, DSI_CMD2_BK0_NVGAMCTRL, 0x00, 0x0E, 0x95, 0x0F,
-		   0x13, 0x07, 0x09, 0x08, 0x08, 0x22, 0x04, 0x10, 0x0E,
-		   0x2C, 0x34, 0x1F);
-	ST7701_DSI(st7701, DSI_CMD2_BK0_LNESET,
-		   DSI_CMD2_BK0_LNESET_B0, DSI_CMD2_BK0_LNESET_B1);
-	ST7701_DSI(st7701, DSI_CMD2_BK0_PORCTRL,
-		   DSI_CMD2_BK0_PORCTRL_B0(mode),
-		   DSI_CMD2_BK0_PORCTRL_B1(mode));
-	ST7701_DSI(st7701, DSI_CMD2_BK0_INVSEL,
-		   DSI_CMD2_BK0_INVSEL_B0, DSI_CMD2_BK0_INVSEL_B1);
-
-	/* Command2, BK1 */
-	ST7701_DSI(st7701, DSI_CMD2BKX_SEL,
-			0x77, 0x01, 0x00, 0x00, DSI_CMD2BK1_SEL);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_VRHS, DSI_CMD2_BK1_VRHA_SET);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_VCOM, DSI_CMD2_BK1_VCOM_SET);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_VGHSS, DSI_CMD2_BK1_VGHSS_SET);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_TESTCMD, DSI_CMD2_BK1_TESTCMD_VAL);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_VGLS, DSI_CMD2_BK1_VGLS_SET);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_PWCTLR1, DSI_CMD2_BK1_PWCTLR1_SET);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_PWCTLR2, DSI_CMD2_BK1_PWCTLR2_SET);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_SPD1, DSI_CMD2_BK1_SPD1_SET);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_SPD2, DSI_CMD2_BK1_SPD2_SET);
-	ST7701_DSI(st7701, DSI_CMD2_BK1_MIPISET1, DSI_CMD2_BK1_MIPISET1_SET);
-
-	/**
-	 * ST7701_SPEC_V1.2 is unable to provide enough information above this
-	 * specific command sequence, so grab the same from vendor BSP driver.
-	 */
-	ST7701_DSI(st7701, 0xE0, 0x00, 0x00, 0x02);
-	ST7701_DSI(st7701, 0xE1, 0x0B, 0x00, 0x0D, 0x00, 0x0C, 0x00, 0x0E,
-		   0x00, 0x00, 0x44, 0x44);
-	ST7701_DSI(st7701, 0xE2, 0x33, 0x33, 0x44, 0x44, 0x64, 0x00, 0x66,
-		   0x00, 0x65, 0x00, 0x67, 0x00, 0x00);
-	ST7701_DSI(st7701, 0xE3, 0x00, 0x00, 0x33, 0x33);
-	ST7701_DSI(st7701, 0xE4, 0x44, 0x44);
-	ST7701_DSI(st7701, 0xE5, 0x0C, 0x78, 0x3C, 0xA0, 0x0E, 0x78, 0x3C,
-		   0xA0, 0x10, 0x78, 0x3C, 0xA0, 0x12, 0x78, 0x3C, 0xA0);
-	ST7701_DSI(st7701, 0xE6, 0x00, 0x00, 0x33, 0x33);
-	ST7701_DSI(st7701, 0xE7, 0x44, 0x44);
-	ST7701_DSI(st7701, 0xE8, 0x0D, 0x78, 0x3C, 0xA0, 0x0F, 0x78, 0x3C,
-		   0xA0, 0x11, 0x78, 0x3C, 0xA0, 0x13, 0x78, 0x3C, 0xA0);
-	ST7701_DSI(st7701, 0xEB, 0x02, 0x02, 0x39, 0x39, 0xEE, 0x44, 0x00);
-	ST7701_DSI(st7701, 0xEC, 0x00, 0x00);
-	ST7701_DSI(st7701, 0xED, 0xFF, 0xF1, 0x04, 0x56, 0x72, 0x3F, 0xFF,
-		   0xFF, 0xFF, 0xFF, 0xF3, 0x27, 0x65, 0x40, 0x1F, 0xFF);
-
-	/* disable Command2 */
-	ST7701_DSI(st7701, DSI_CMD2BKX_SEL,
-		   0x77, 0x01, 0x00, 0x00, DSI_CMD2BKX_SEL_NONE);
+	struct st7701s_panel *st7701s = to_st7701s_panel(panel);
+
+	if (st7701s->prepared)
+		return 0;
+
+	if (st7701s->reset != NULL) {
+		gpiod_set_value(st7701s->reset, 1);
+		mdelay(20);
+		gpiod_set_value(st7701s->reset, 0);
+		mdelay(10);
+		gpiod_set_value(st7701s->reset, 1);
+		mdelay(120);
+	}
+
+	st7701s->prepared = true;
+
+	return 0;
 }
 
-static int st7701_prepare(struct drm_panel *panel)
+static int st7701s_panel_unprepare(struct drm_panel *panel)
 {
-	struct st7701 *st7701 = panel_to_st7701(panel);
-	int ret;
+	struct st7701s_panel *st7701s = to_st7701s_panel(panel);
+	struct device *dev = &st7701s->dsi->dev;
 
-	gpiod_set_value(st7701->reset, 0);
+	if (!st7701s->prepared)
+		return 0;
 
-	ret = regulator_bulk_enable(st7701->desc->num_supplies,
-				    st7701->supplies);
-	if (ret < 0)
-		return ret;
-	msleep(20);
+	if (st7701s->enabled) {
+		DRM_DEV_ERROR(dev, "Panel still enabled!\n");
+		return -EPERM;
+	}
 
-	gpiod_set_value(st7701->reset, 1);
-	msleep(150);
+	if (st7701s->reset != NULL) {
+		gpiod_set_value(st7701s->reset, 0);
+		mdelay(1);
+		gpiod_set_value(st7701s->reset, 1);
+	}
 
-	st7701_init_sequence(st7701);
+	st7701s->prepared = false;
 
 	return 0;
 }
 
-static int st7701_enable(struct drm_panel *panel)
+static int st7701s_panel_enable(struct drm_panel *panel)
 {
-	struct st7701 *st7701 = panel_to_st7701(panel);
+	struct st7701s_panel *st7701s = to_st7701s_panel(panel);
+	struct mipi_dsi_device *dsi = st7701s->dsi;
+	struct device *dev = &dsi->dev;
+	int ret;
 
-	ST7701_DSI(st7701, MIPI_DCS_SET_DISPLAY_ON, 0x00);
+	if (st7701s->enabled)
+		return 0;
 
-	return 0;
-}
+	if (!st7701s->prepared) {
+		DRM_DEV_ERROR(dev, "Panel not prepared!\n");
+		return -EPERM;
+	}
 
-static int st7701_disable(struct drm_panel *panel)
-{
-	struct st7701 *st7701 = panel_to_st7701(panel);
+	dsi->mode_flags |= MIPI_DSI_MODE_LPM;
+
+	ret = mipi_dsi_dcs_exit_sleep_mode(dsi);
+	if (ret < 0) {
+		DRM_DEV_ERROR(dev, "Failed to exit sleep mode (%d)\n", ret);
+		goto fail;
+	}
+
+	mdelay(120);
+
+	ret = st7701s_panel_push_cmd_list(dsi);
+	if (ret < 0) {
+		DRM_DEV_ERROR(dev, "Failed to send init cmds (%d)\n", ret);
+		goto fail;
+	}
+
+	mdelay(5);
 
-	ST7701_DSI(st7701, MIPI_DCS_SET_DISPLAY_OFF, 0x00);
+	ret = mipi_dsi_dcs_set_display_on(dsi);
+	if (ret < 0) {
+		DRM_DEV_ERROR(dev, "Failed to set display ON (%d)\n", ret);
+		goto fail;
+	}
+
+	st7701s->backlight->props.power = FB_BLANK_UNBLANK;
+	backlight_update_status(st7701s->backlight);
+
+	st7701s->enabled = true;
 
 	return 0;
+
+fail:
+	if (st7701s->reset != NULL)
+		gpiod_set_value(st7701s->reset, 0);
+
+	return ret;
 }
 
-static int st7701_unprepare(struct drm_panel *panel)
+static int st7701s_panel_disable(struct drm_panel *panel)
 {
-	struct st7701 *st7701 = panel_to_st7701(panel);
+	struct st7701s_panel *st7701s = to_st7701s_panel(panel);
+	struct mipi_dsi_device *dsi = st7701s->dsi;
+	struct device *dev = &dsi->dev;
+	int ret;
 
-	ST7701_DSI(st7701, MIPI_DCS_ENTER_SLEEP_MODE, 0x00);
+	if (!st7701s->enabled)
+		return 0;
 
-	msleep(st7701->sleep_delay);
+	dsi->mode_flags |= MIPI_DSI_MODE_LPM;
 
-	gpiod_set_value(st7701->reset, 0);
+	ret = mipi_dsi_dcs_set_display_off(dsi);
+	if (ret < 0) {
+		DRM_DEV_ERROR(dev, "Failed to set display OFF (%d)\n", ret);
+		return ret;
+	}
 
-	/**
-	 * During the Resetting period, the display will be blanked
-	 * (The display is entering blanking sequence, which maximum
-	 * time is 120 ms, when Reset Starts in Sleep Out –mode. The
-	 * display remains the blank state in Sleep In –mode.) and
-	 * then return to Default condition for Hardware Reset.
-	 *
-	 * So we need wait sleep_delay time to make sure reset completed.
-	 */
-	msleep(st7701->sleep_delay);
+	usleep_range(5000, 10000);
+
+	ret = mipi_dsi_dcs_enter_sleep_mode(dsi);
+	if (ret < 0) {
+		DRM_DEV_ERROR(dev, "Failed to enter sleep mode (%d)\n", ret);
+		return ret;
+	}
+
+	usleep_range(10000, 15000);
 
-	regulator_bulk_disable(st7701->desc->num_supplies, st7701->supplies);
+	st7701s->backlight->props.power = FB_BLANK_POWERDOWN;
+	backlight_update_status(st7701s->backlight);
+
+	st7701s->enabled = false;
 
 	return 0;
 }
 
-static int st7701_get_modes(struct drm_panel *panel,
-			    struct drm_connector *connector)
+static int st7701s_panel_get_modes(struct drm_panel *panel,
+					struct drm_connector *connector)
 {
-	struct st7701 *st7701 = panel_to_st7701(panel);
-	const struct drm_display_mode *desc_mode = st7701->desc->mode;
+	struct st7701s_panel *st7701s = to_st7701s_panel(panel);
+	struct device *dev = &st7701s->dsi->dev;
+	connector->connector_type = panel->connector_type;
 	struct drm_display_mode *mode;
+	u32 *bus_flags = &connector->display_info.bus_flags;
+	int ret;
 
-	mode = drm_mode_duplicate(connector->dev, desc_mode);
+	mode = drm_mode_create(connector->dev);
 	if (!mode) {
-		dev_err(&st7701->dsi->dev, "failed to add mode %ux%u@%u\n",
-			desc_mode->hdisplay, desc_mode->vdisplay,
-			drm_mode_vrefresh(desc_mode));
-		return -ENOMEM;
+		DRM_DEV_ERROR(dev, "Failed to create display mode!\n");
+		return 0;
 	}
 
-	drm_mode_set_name(mode);
-	drm_mode_probed_add(connector, mode);
+	drm_display_mode_from_videomode(&st7701s->vm, mode);
+	mode->width_mm = st7701s->width_mm;
+	mode->height_mm = st7701s->height_mm;
+	connector->display_info.width_mm = st7701s->width_mm;
+	connector->display_info.height_mm = st7701s->height_mm;
+	mode->type = DRM_MODE_TYPE_DRIVER | DRM_MODE_TYPE_PREFERRED;
+
+	if (st7701s->vm.flags & DISPLAY_FLAGS_DE_HIGH)
+		*bus_flags |= DRM_BUS_FLAG_DE_HIGH;
+	if (st7701s->vm.flags & DISPLAY_FLAGS_DE_LOW)
+		*bus_flags |= DRM_BUS_FLAG_DE_LOW;
+	if (st7701s->vm.flags & DISPLAY_FLAGS_PIXDATA_NEGEDGE)
+		*bus_flags |= DRM_BUS_FLAG_PIXDATA_DRIVE_NEGEDGE;
+	if (st7701s->vm.flags & DISPLAY_FLAGS_PIXDATA_POSEDGE)
+		*bus_flags |= DRM_BUS_FLAG_PIXDATA_DRIVE_POSEDGE;
+
+	ret = drm_display_info_set_bus_formats(&connector->display_info,
+			st7701s_bus_formats, ARRAY_SIZE(st7701s_bus_formats));
+	if (ret)
+		return ret;
 
-	connector->display_info.width_mm = desc_mode->width_mm;
-	connector->display_info.height_mm = desc_mode->height_mm;
+	drm_mode_probed_add(connector, mode);
 
 	return 1;
 }
 
-static const struct drm_panel_funcs st7701_funcs = {
-	.disable	= st7701_disable,
-	.unprepare	= st7701_unprepare,
-	.prepare	= st7701_prepare,
-	.enable		= st7701_enable,
-	.get_modes	= st7701_get_modes,
-};
+static int st7701s_bl_get_brightness(struct backlight_device *bl)
+{
+	struct mipi_dsi_device *dsi = bl_get_data(bl);
+	struct st7701s_panel *st7701s = mipi_dsi_get_drvdata(dsi);
+	struct device *dev = &dsi->dev;
+	u16 brightness;
+	int ret;
+
+	if (!st7701s->prepared)
+		return 0;
+
+	DRM_DEV_DEBUG_DRIVER(dev, "\n");
+
+	dsi->mode_flags &= ~MIPI_DSI_MODE_LPM;
 
-static const struct drm_display_mode ts8550b_mode = {
-	.clock		= 27500,
+	ret = mipi_dsi_dcs_get_display_brightness(dsi, &brightness);
+	if (ret < 0)
+		return ret;
+
+	bl->props.brightness = brightness;
+
+	return brightness & 0xff;
+}
+
+static int st7701s_bl_update_status(struct backlight_device *bl)
+{
+	struct mipi_dsi_device *dsi = bl_get_data(bl);
+	struct st7701s_panel *st7701s = mipi_dsi_get_drvdata(dsi);
+	struct device *dev = &dsi->dev;
+	int ret = 0;
 
-	.hdisplay	= 480,
-	.hsync_start	= 480 + 38,
-	.hsync_end	= 480 + 38 + 12,
-	.htotal		= 480 + 38 + 12 + 12,
+	if (!st7701s->prepared)
+		return 0;
 
-	.vdisplay	= 854,
-	.vsync_start	= 854 + 18,
-	.vsync_end	= 854 + 18 + 8,
-	.vtotal		= 854 + 18 + 8 + 4,
+	DRM_DEV_DEBUG_DRIVER(dev, "New brightness: %d\n", bl->props.brightness);
 
-	.width_mm	= 69,
-	.height_mm	= 139,
+	dsi->mode_flags &= ~MIPI_DSI_MODE_LPM;
+
+	ret = mipi_dsi_dcs_set_display_brightness(dsi, bl->props.brightness);
+	if (ret < 0)
+		return ret;
+
+	return 0;
+}
 
-	.type = DRM_MODE_TYPE_DRIVER | DRM_MODE_TYPE_PREFERRED,
+static const struct backlight_ops st7701s_bl_ops = {
+	.update_status = st7701s_bl_update_status,
+	.get_brightness = st7701s_bl_get_brightness,
 };
 
-static const char * const ts8550b_supply_names[] = {
-	"VCC",
-	"IOVCC",
+static const struct drm_panel_funcs st7701s_panel_funcs = {
+	.prepare = st7701s_panel_prepare,
+	.unprepare = st7701s_panel_unprepare,
+	.enable = st7701s_panel_enable,
+	.disable = st7701s_panel_disable,
+	.get_modes = st7701s_panel_get_modes,
 };
 
-static const struct st7701_panel_desc ts8550b_desc = {
-	.mode = &ts8550b_mode,
-	.lanes = 2,
-	.flags = MIPI_DSI_MODE_VIDEO,
-	.format = MIPI_DSI_FMT_RGB888,
-	.supply_names = ts8550b_supply_names,
-	.num_supplies = ARRAY_SIZE(ts8550b_supply_names),
-	.panel_sleep_delay = 80, /* panel need extra 80ms for sleep out cmd */
+/*
+ * The clock might range from 66MHz (30Hz refresh rate)
+ * to 132MHz (60Hz refresh rate)
+ */
+static const struct display_timing st7701s_default_timing = {
+	.pixelclock = {8294400, 16588800, 16588800}, //Hz : htotal*vtotal*60 = 23 908 800 Hz
+	.hactive = { 480, 480, 480 },
+	.hfront_porch = { 1, 1, 50 },
+	.hsync_len = { 1, 1, 10 },
+	.hback_porch = { 50, 50, 46 },
+	.vactive = { 480, 480, 480 },
+	.vfront_porch = { 10, 15, 15 },
+	.vsync_len = { 1, 1, 10 },
+	.vback_porch = { 5, 15, 15 },
+	.flags = DISPLAY_FLAGS_HSYNC_LOW |
+		 DISPLAY_FLAGS_VSYNC_LOW |
+		 DISPLAY_FLAGS_DE_LOW|
+		 DISPLAY_FLAGS_PIXDATA_NEGEDGE,
 };
 
-static int st7701_dsi_probe(struct mipi_dsi_device *dsi)
+static int st7701s_panel_probe(struct mipi_dsi_device *dsi)
 {
-	const struct st7701_panel_desc *desc;
-	struct st7701 *st7701;
-	int ret, i;
 
-	st7701 = devm_kzalloc(&dsi->dev, sizeof(*st7701), GFP_KERNEL);
-	if (!st7701)
+	struct device *dev = &dsi->dev;
+	struct device_node *np = dev->of_node;
+	struct device_node *timings;
+	struct st7701s_panel *panel;
+	struct backlight_properties bl_props;
+	int ret;
+	u32 video_mode;
+	panel = devm_kzalloc(&dsi->dev, sizeof(*panel), GFP_KERNEL);
+	if (!panel)
 		return -ENOMEM;
 
-	desc = of_device_get_match_data(&dsi->dev);
-	dsi->mode_flags = desc->flags;
-	dsi->format = desc->format;
-	dsi->lanes = desc->lanes;
-
-	st7701->supplies = devm_kcalloc(&dsi->dev, desc->num_supplies,
-					sizeof(*st7701->supplies),
-					GFP_KERNEL);
-	if (!st7701->supplies)
-		return -ENOMEM;
+	mipi_dsi_set_drvdata(dsi, panel);
+
+	panel->dsi = dsi;
+
+	dsi->format = MIPI_DSI_FMT_RGB888;
+	dsi->mode_flags =  MIPI_DSI_MODE_VIDEO_HSE | MIPI_DSI_MODE_VIDEO |
+			   MIPI_DSI_CLOCK_NON_CONTINUOUS;
+
+	ret = of_property_read_u32(np, "video-mode", &video_mode);
+	if (!ret) {
+		switch (video_mode) {
+		case 0:
+			/* burst mode */
+			dsi->mode_flags |= MIPI_DSI_MODE_VIDEO_BURST;
+			break;
+		case 1:
+			/* non-burst mode with sync event */
+			break;
+		case 2:
+			/* non-burst mode with sync pulse */
+			dsi->mode_flags |= MIPI_DSI_MODE_VIDEO_SYNC_PULSE;
+			break;
+		default:
+			dev_warn(dev, "invalid video mode %d\n", video_mode);
+			break;
+
+		}
+	}
 
-	for (i = 0; i < desc->num_supplies; i++)
-		st7701->supplies[i].supply = desc->supply_names[i];
+	ret = of_property_read_u32(np, "dsi-lanes", &dsi->lanes);
+	if (ret < 0) {
+		dev_err(dev, "Failed to get dsi-lanes property (%d)\n", ret);
+		return ret;
+	}
 
-	ret = devm_regulator_bulk_get(&dsi->dev, desc->num_supplies,
-				      st7701->supplies);
+	/*
+	 * 'display-timings' is optional, so verify if the node is present
+	 * before calling of_get_videomode so we won't get console error
+	 * messages
+	 */
+	timings = of_get_child_by_name(np, "display-timings");
+	if (timings) {
+		of_node_put(timings);
+		ret = of_get_videomode(np, &panel->vm, 0);
+	} else {
+		videomode_from_timing(&st7701s_default_timing, &panel->vm);
+	}
 	if (ret < 0)
 		return ret;
 
-	st7701->reset = devm_gpiod_get(&dsi->dev, "reset", GPIOD_OUT_LOW);
-	if (IS_ERR(st7701->reset)) {
-		dev_err(&dsi->dev, "Couldn't get our reset GPIO\n");
-		return PTR_ERR(st7701->reset);
-	}
+	of_property_read_u32(np, "panel-width-mm", &panel->width_mm);
+	of_property_read_u32(np, "panel-height-mm", &panel->height_mm);
 
-	drm_panel_init(&st7701->panel, &dsi->dev, &st7701_funcs,
-		       DRM_MODE_CONNECTOR_DSI);
-
-	/**
-	 * Once sleep out has been issued, ST7701 IC required to wait 120ms
-	 * before initiating new commands.
-	 *
-	 * On top of that some panels might need an extra delay to wait, so
-	 * add panel specific delay for those cases. As now this panel specific
-	 * delay information is referenced from those panel BSP driver, example
-	 * ts8550b and there is no valid documentation for that.
-	 */
-	st7701->sleep_delay = 120 + desc->panel_sleep_delay;
+	panel->reset = devm_gpiod_get(dev, "reset", GPIOD_OUT_HIGH);
 
-	ret = drm_panel_of_backlight(&st7701->panel);
-	if (ret)
+	if (IS_ERR(panel->reset)) {
+		panel->reset = NULL;
+	}
+	else
+		gpiod_set_value(panel->reset, 0);
+
+
+	memset(&bl_props, 0, sizeof(bl_props));
+	bl_props.type = BACKLIGHT_RAW;
+	bl_props.brightness = 255;
+	bl_props.max_brightness = 255;
+
+	panel->backlight = devm_backlight_device_register(
+				dev, dev_name(dev),
+				dev, dsi,
+				&st7701s_bl_ops, &bl_props);
+	if (IS_ERR(panel->backlight)) {
+		ret = PTR_ERR(panel->backlight);
+		dev_err(dev, "Failed to register backlight (%d)\n", ret);
 		return ret;
+	}
 
-	drm_panel_add(&st7701->panel);
+	drm_panel_init(&panel->base, &dsi->dev, &st7701s_panel_funcs,
+			DRM_MODE_CONNECTOR_DSI);
+	panel->base.funcs = &st7701s_panel_funcs;
+	panel->base.dev = dev;
 
-	mipi_dsi_set_drvdata(dsi, st7701);
-	st7701->dsi = dsi;
-	st7701->desc = desc;
+	drm_panel_add(&panel->base);
 
-	return mipi_dsi_attach(dsi);
+	ret = mipi_dsi_attach(dsi);
+	if (ret < 0)
+	{
+		drm_panel_remove(&panel->base);
+	}
+	return ret;
 }
 
-static int st7701_dsi_remove(struct mipi_dsi_device *dsi)
+static int st7701s_panel_remove(struct mipi_dsi_device *dsi)
 {
-	struct st7701 *st7701 = mipi_dsi_get_drvdata(dsi);
+	struct st7701s_panel *st7701s = mipi_dsi_get_drvdata(dsi);
+	struct device *dev = &dsi->dev;
+	int ret;
+
+	ret = mipi_dsi_detach(dsi);
+	if (ret < 0)
+		DRM_DEV_ERROR(dev, "Failed to detach from host (%d)\n",
+			ret);
 
-	mipi_dsi_detach(dsi);
-	drm_panel_remove(&st7701->panel);
+	drm_panel_disable(&st7701s->base);
+
+	if (st7701s->base.dev)
+		drm_panel_remove(&st7701s->base);
 
 	return 0;
 }
 
-static const struct of_device_id st7701_of_match[] = {
-	{ .compatible = "techstar,ts8550b", .data = &ts8550b_desc },
+static void st7701s_panel_shutdown(struct mipi_dsi_device *dsi)
+{
+	struct st7701s_panel *st7701s= mipi_dsi_get_drvdata(dsi);
+
+	st7701s_panel_disable(&st7701s->base);
+	st7701s_panel_unprepare(&st7701s->base);//some panel do not need;
+}
+
+static const struct of_device_id st7701s_of_match[] = {
+	{ .compatible = "sitronix,st7701s", },
 	{ }
 };
-MODULE_DEVICE_TABLE(of, st7701_of_match);
+MODULE_DEVICE_TABLE(of, st7701s_of_match);
 
-static struct mipi_dsi_driver st7701_dsi_driver = {
-	.probe		= st7701_dsi_probe,
-	.remove		= st7701_dsi_remove,
+static struct mipi_dsi_driver st7701s_panel_driver = {
 	.driver = {
-		.name		= "st7701",
-		.of_match_table	= st7701_of_match,
+		.name = "panel-sirtonix-st7701s",
+		.of_match_table = st7701s_of_match,
 	},
+	.probe = st7701s_panel_probe,
+	.remove = st7701s_panel_remove,
+	.shutdown = st7701s_panel_shutdown,
 };
-module_mipi_dsi_driver(st7701_dsi_driver);
+module_mipi_dsi_driver(st7701s_panel_driver);
+
+MODULE_AUTHOR("Fangkai <fangkai@kyee.com.cn>");
+MODULE_DESCRIPTION("sitronix st7701s lcd driver");
+MODULE_LICENSE("GPL v2");
 
-MODULE_AUTHOR("Jagan Teki <jagan@amarulasolutions.com>");
-MODULE_DESCRIPTION("Sitronix ST7701 LCD Panel Driver");
-MODULE_LICENSE("GPL");
-- 
2.25.1

