From 4871e3ba53d64664f3a71e8808a69b66d3bbc70f Mon Sep 17 00:00:00 2001
From: chiragp-mechas <chiragp@mechasystems.com>
Date: Fri, 1 Dec 2023 18:04:24 +0530
Subject: [PATCH] Fix-default-gpio-configuration-mickledore

---
 drivers/gpio/Kconfig | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpio/Kconfig b/drivers/gpio/Kconfig
index 018bf9a98f0b..c4be3abc35fb 100644
--- a/drivers/gpio/Kconfig
+++ b/drivers/gpio/Kconfig
@@ -3,6 +3,9 @@
 # GPIO infrastructure and drivers
 #
 
+config GPIO_SYSFS
+	default y
+
 config ARCH_HAVE_CUSTOM_GPIO_H
 	bool
 	help
-- 
2.25.1

