From 53a5c7d43d96ab4ce151ce715cdcd68e2a22dad7 Mon Sep 17 00:00:00 2001
From: chiragp-mechas <chiragp@mechasystems.com>
Date: Mon, 8 Jan 2024 14:09:25 +0530
Subject: [PATCH] Resolve-camera-invert-preview-issue

---
 .../platform/mxc/capture/ov5640_mipi_v2.c     | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c b/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c
index cdac554ddf82..80bf1b9db360 100644
--- a/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c
+++ b/drivers/media/platform/mxc/capture/ov5640_mipi_v2.c
@@ -47,6 +47,11 @@
 
 #define DEFAULT_SCCB_ID 0x78
 
+#define OV5640_TIMING_TC_REG20		0x3820
+#define OV5640_TIMING_TC_REG20_VFLIP	0x06
+#define OV5640_TIMING_TC_REG21		0x3821
+#define OV5640_TIMING_TC_REG21_MIRROR	0x06
+
 enum ov5640_mode {
 	ov5640_mode_MIN = 0,
 	ov5640_mode_VGA_640_480 = 0,
@@ -115,6 +120,8 @@ struct ov5640 {
 	int green;
 	int blue;
 	int ae_mode;
+	int vflip;
+	int mirror;
 
 	u32 mclk;
 	u8 mclk_source;
@@ -1012,6 +1019,8 @@ static int ov5640_download_firmware(struct ov5640 *sensor,
 	register u8 Val = 0;
 	u8 RegVal = 0;
 	int i, retval = 0;
+	sensor->vflip = 0;
+	//sensor->mirror = 0;
 
 	for (i = 0; i < ArySize; ++i, ++pModeSetting) {
 		Delay_ms = pModeSetting->u32Delay_ms;
@@ -1029,6 +1038,24 @@ static int ov5640_download_firmware(struct ov5640 *sensor,
 			Val |= RegVal;
 		}
 
+		/* Overwrite vflip value if provided in device tree */
+		if ((RegAddr == OV5640_TIMING_TC_REG20) &&
+		    (sensor->vflip != -1)) {
+			if (sensor->vflip)
+				Val &= ~(OV5640_TIMING_TC_REG20_VFLIP);
+			else
+				Val |= (OV5640_TIMING_TC_REG20_VFLIP);
+		}
+
+		/* Overwrite mirror value if provided in device tree */
+		if ((RegAddr == OV5640_TIMING_TC_REG21) &&
+		    (sensor->mirror != -1)) {
+			if (sensor->mirror)
+				Val |= (OV5640_TIMING_TC_REG21_MIRROR);
+			else
+				Val &= ~(OV5640_TIMING_TC_REG21_MIRROR);
+		}
+
 		retval = ov5640_write_reg(sensor, RegAddr, Val);
 		if (retval < 0)
 			goto err;
@@ -1765,6 +1792,15 @@ static int ov5640_probe(struct i2c_client *client,
 		return retval;
 	}
 
+	/* Get optional mirror/vflip values */
+	retval = of_property_read_u32(dev->of_node, "mirror", &sensor->mirror);
+	if (retval)
+		sensor->mirror = ~(-1);
+
+	retval = of_property_read_u32(dev->of_node, "vflip", &sensor->vflip);
+	if (retval)
+		sensor->vflip= -1;
+
 	clk_prepare_enable(sensor->sensor_clk);
 
 	sensor->io_init = ov5640_reset;
-- 
2.25.1

