diff --git a/drivers/gpu/drm/etnaviv/etnaviv_drv.c b/drivers/gpu/drm/etnaviv/etnaviv_drv.c
index 7dcc6392792d..afd316631c1e 100644
--- a/drivers/gpu/drm/etnaviv/etnaviv_drv.c
+++ b/drivers/gpu/drm/etnaviv/etnaviv_drv.c
@@ -590,8 +590,9 @@ static int etnaviv_pdev_probe(struct platform_device *pdev)
 {
 	struct device *dev = &pdev->dev;
 	struct component_match *match = NULL;
-
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 	if (!dev->platform_data) {
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 		struct device_node *core_node;
 
 		for_each_compatible_node(core_node, NULL, "vivante,gc") {
@@ -602,6 +603,7 @@ static int etnaviv_pdev_probe(struct platform_device *pdev)
 						   compare_of, core_node);
 		}
 	} else {
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 		char **names = dev->platform_data;
 		unsigned i;
 
@@ -609,6 +611,7 @@ static int etnaviv_pdev_probe(struct platform_device *pdev)
 			component_match_add(dev, &match, compare_str, names[i]);
 	}
 
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 	return component_master_add_with_match(dev, &etnaviv_master_ops, match);
 }
 
@@ -637,24 +640,29 @@ static int __init etnaviv_init(void)
 
 	etnaviv_validate_init();
 
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 	ret = platform_driver_register(&etnaviv_gpu_driver);
 	if (ret != 0)
 		return ret;
 
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 	ret = platform_driver_register(&etnaviv_platform_driver);
 	if (ret != 0)
 		goto unregister_gpu_driver;
 
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 	/*
 	 * If the DT contains at least one available GPU device, instantiate
 	 * the DRM platform device.
 	 */
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 	for_each_compatible_node(np, NULL, "vivante,gc") {
 		if (!of_device_is_available(np))
 			continue;
 
 		pdev = platform_device_alloc("etnaviv", -1);
 		if (!pdev) {
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 			ret = -ENOMEM;
 			of_node_put(np);
 			goto unregister_platform_driver;
@@ -671,6 +679,7 @@ static int __init etnaviv_init(void)
 
 		ret = platform_device_add(pdev);
 		if (ret) {
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 			platform_device_put(pdev);
 			of_node_put(np);
 			goto unregister_platform_driver;
@@ -680,12 +689,15 @@ static int __init etnaviv_init(void)
 		of_node_put(np);
 		break;
 	}
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 
 	return 0;
 
 unregister_platform_driver:
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 	platform_driver_unregister(&etnaviv_platform_driver);
 unregister_gpu_driver:
+	printk("EDEBUG: %s-%d\n",__func__,__LINE__);
 	platform_driver_unregister(&etnaviv_gpu_driver);
 	return ret;
 }
