From 485399047f5de9ab036298c80d6eac3f1120dd6f Mon Sep 17 00:00:00 2001
From: Hardevsinh Palaniya <hardevsinh.palaniya@siliconsignals.io>
Date: Mon, 25 Sep 2023 19:00:59 +0530
Subject: [PATCH] GPU with 4GB RAM Dts Changes Working

Signed-off-by: Hardevsinh Palaniya <hardevsinh.palaniya@siliconsignals.io>

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-demo.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-demo.dts
index fefabd489952..3b715a6c044b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-demo.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-demo.dts
@@ -552,6 +552,11 @@ &vpu_h1 {
 	status = "okay";
 };
 
-&gpu {
-	status = "okay";
-};
+//&gpu {
+//        //reg = <0x0 0x0 0x0 0x8000000>;
+//	reg = <0x0 0x38000000 0x0 0x8000>, <0x0 0x38008000 0x0 0x8000>,
+//        <0x0 0x40000000 0x0 0x80000000>, <0x0 0x38000000 0x0 0x40000>;
+//        //reg = <0x0 0x38000000 0x0 0x40000>;
+//        //reg-names = "contiguous_mem";
+//	status = "okay";
+//};
diff --git a/arch/arm64/boot/dts/freescale/imx8mm.dtsi b/arch/arm64/boot/dts/freescale/imx8mm.dtsi
index 177efc39ddb3..1254814793a7 100755
--- a/arch/arm64/boot/dts/freescale/imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mm.dtsi
@@ -1497,11 +1497,13 @@ vpu_v4l2: vpu_v4l2 {
 	};
 
         gpu: gpu@38000000 {
-		compatible ="fsl,imx8mm-gpu", "fsl,imx6q-gpu", "vivante,gc";
-		reg = <0x0 0x38000000 0x0 0x8000>, <0x0 0x38008000 0x0 0x8000>,
-			<0x0 0x40000000 0x0 0x80000000>, <0x0 0x0 0x0 0x8000000>;
-		reg-names = "iobase_3d", "iobase_2d",
-			"phys_baseaddr", "contiguous_mem";
+		//compatible ="fsl,imx8mm-gpu", "fsl,imx6q-gpu", "vivante,gc";
+		//reg = <0x0 0x38000000 0x0 0x8000>, <0x0 0x38008000 0x0 0x8000>,
+		//	<0x0 0x40000000 0x0 0x80000000>, <0x0 0x0 0x0 0x8000000>;
+	        compatible = "vivante,gc";
+		reg = <0x0 0x38000000 0x0 0x8000>;
+		//reg-names = "iobase_3d", "iobase_2d",
+		//	"phys_baseaddr", "contiguous_mem";
 		interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>,
 			<GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "irq_3d", "irq_2d";
@@ -1512,10 +1514,11 @@ gpu: gpu@38000000 {
 			<&clk IMX8MM_CLK_GPU2D_ROOT>,
 			<&clk IMX8MM_CLK_GPU_BUS_ROOT>,
 			<&clk IMX8MM_CLK_GPU_AHB>;
-		clock-names = "core", "gpu3d_shader_clk",
-			  "gpu3d_axi_clk", "gpu3d_ahb_clk",
-			  "gpu2d_clk", "gpu2d_axi_clk",
-			  "gpu2d_ahb_clk";
+		clock-names = "core", "shader", "bus", "reg";
+		//clock-names = "core", "gpu3d_shader_clk",
+		//	  "gpu3d_axi_clk", "gpu3d_ahb_clk",
+		//	  "gpu2d_clk", "gpu2d_axi_clk",
+		//	  "gpu2d_ahb_clk";
 		assigned-clocks = <&clk IMX8MM_CLK_GPU3D_SRC>,
 				<&clk IMX8MM_CLK_GPU2D_SRC>,
 				<&clk IMX8MM_CLK_GPU_AXI>,
diff --git a/arch/arm64/boot/dts/freescale/mecha-mc-ci8mm.dts b/arch/arm64/boot/dts/freescale/mecha-mc-ci8mm.dts
index 243044e7223e..24679194a759 100644
--- a/arch/arm64/boot/dts/freescale/mecha-mc-ci8mm.dts
+++ b/arch/arm64/boot/dts/freescale/mecha-mc-ci8mm.dts
@@ -457,3 +457,12 @@ &pwm3 {
         status = "okay";
 };
 
+&gpu {
+        //reg = <0x0 0x0 0x0 0x8000000>;
+        reg = <0x0 0x38000000 0x0 0x8000>, <0x0 0x38008000 0x0 0x8000>,
+        <0x0 0x40000000 0x0 0x80000000>, <0x0 0x38000000 0x0 0x40000>;
+        //reg = <0x0 0x38000000 0x0 0x40000>;
+        //reg-names = "contiguous_mem";
+        status = "okay";
+};
+
-- 
2.25.1

