diff --git a/board/freescale/imx8mm_evk/imx8mm_evk.c b/board/freescale/imx8mm_evk/imx8mm_evk.c
index 89ff705edf..936d12b431 100644
--- a/board/freescale/imx8mm_evk/imx8mm_evk.c
+++ b/board/freescale/imx8mm_evk/imx8mm_evk.c
@@ -24,6 +24,7 @@
 #include <imx_sip.h>
 #include <linux/arm-smccc.h>
 
+#define DEVICE_SHUTDOWN_BATTERY_PERCENT 10
 DECLARE_GLOBAL_DATA_PTR;
 
 #define UART_PAD_CTRL	(PAD_CTL_DSE6 | PAD_CTL_FSEL1)
@@ -168,9 +169,7 @@ int power_fg_i2c_init_update(uint8_t i2c_bus, uint8_t addr)
 		printf("Err:i2c bus handler not found\n");
 		return -ENODEV;
 	}
-	power_fg_init_update(i2c_bus, i2c_dev);
-
-	return 0;
+	return power_fg_init_update(i2c_bus, i2c_dev);
 }
 
 static int setup_pd_switch(uint8_t i2c_bus, uint8_t addr)
@@ -345,6 +344,7 @@ int board_ehci_usb_phy_mode(struct udevice *dev)
 
 #define DISPMIX				9
 #define MIPI				10
+bool is_battery_low = false;
 
 int board_init(void)
 {
@@ -357,21 +357,39 @@ int board_init(void)
 	if (IS_ENABLED(CONFIG_FEC_MXC))
 		setup_fec();
 
-	arm_smccc_smc(IMX_SIP_GPC, IMX_SIP_GPC_PM_DOMAIN,
-		      DISPMIX, true, 0, 0, 0, 0, &res);
-	arm_smccc_smc(IMX_SIP_GPC, IMX_SIP_GPC_PM_DOMAIN,
-		      MIPI, true, 0, 0, 0, 0, &res);
+	if (!(power_fg_i2c_init_update(2, 0x55) < 0)) {
+		bq27xxx_battery *bat;
+		power_read_battery_property(bat);
+		printf("Battery : persentage %d\n", bat->soc);
+		if (bat->soc < DEVICE_SHUTDOWN_BATTERY_PERCENT && bat->supply_status != POWER_SUPPLY_STATUS_CHARGING) {
+			is_battery_low = true;
+			printf("Battery : Lower then 10 persentage and charger hasn't connected\n");
+			arm_smccc_smc(IMX_SIP_GPC, IMX_SIP_GPC_PM_DOMAIN, DISPMIX, true, 0, 0, 0, 0, &res);
+			arm_smccc_smc(IMX_SIP_GPC, IMX_SIP_GPC_PM_DOMAIN, MIPI, true, 0, 0, 0, 0, &res);
+		} else {
+			arm_smccc_smc(IMX_SIP_GPC, IMX_SIP_GPC_PM_DOMAIN, DISPMIX, true, 0, 0, 0, 0, &res);
+			arm_smccc_smc(IMX_SIP_GPC, IMX_SIP_GPC_PM_DOMAIN, MIPI, false, 0, 0, 0, 0, &res);
+		}
+	} else {
+		arm_smccc_smc(IMX_SIP_GPC, IMX_SIP_GPC_PM_DOMAIN, DISPMIX, true, 0, 0, 0, 0, &res);
+		arm_smccc_smc(IMX_SIP_GPC, IMX_SIP_GPC_PM_DOMAIN, MIPI, false, 0, 0, 0, 0, &res);
+	}
 
 	return 0;
 }
 
 int board_late_init(void)
 {
+	if (is_battery_low) {
+		printf("Power Down\n");
+		mdelay(2000);
+		memset(0x30370038, 0x60, 1);
+	}
+
 #ifdef CONFIG_ENV_IS_IN_MMC
 	board_late_mmc_env_init();
 #endif
 	
-	power_fg_i2c_init_update(2, 0x55);
 	if (IS_ENABLED(CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG)) {
 		env_set("board_name", "EVK");
 		env_set("board_rev", "iMX8MM");
